vcd/resource_vcd_lb_virtual_server_test.go

```diff
@@ -0,0 +1,258 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckVcdLbVirtualServerDestroy(params["VirtualServerName"].(string)),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_lb_virtual_server.http", "id", regexp.MustCompile(`^virtualServer-\d*$`)),
+					resource.TestCheckResourceAttr("vcd_lb_virtual_server.http", "name", params["VirtualServerName"].(string)),
+					resource.TestCheckResourceAttr("vcd_lb_virtual_server.http", "ip_address", params["EdgeGatewayIp"].(string)),
+					resource.TestCheckResourceAttr("vcd_lb_virtual_server.http", "protocol", "http"),
+					resource.TestCheckResourceAttr("vcd_lb_virtual_server.http", "port", "8888"),
+					resource.TestMatchResourceAttr("vcd_lb_virtual_server.http", "app_profile_id", regexp.MustCompile(`^applicationProfile-\d*$`)),
+					resource.TestMatchResourceAttr("vcd_lb_virtual_server.http", "server_pool_id", regexp.MustCompile(`^pool-\d*$`)),
+					resource.TestCheckResourceAttr("vcd_lb_virtual_server.http", "app_rule_ids.#", "2"),
+					resource.TestMatchResourceAttr("vcd_lb_virtual_server.http", "app_rule_ids.0", regexp.MustCompile(`^applicationRule-\d*$`)),
+					resource.TestMatchResourceAttr("vcd_lb_virtual_server.http", "app_rule_ids.1", regexp.MustCompile(`^applicationRule-\d*$`)),
+
+					// Data source
+					resource.TestMatchResourceAttr("data.vcd_lb_virtual_server.http", "id", regexp.MustCompile(`^virtualServer-\d*$`)),
+					resource.TestCheckResourceAttr("data.vcd_lb_virtual_server.http", "name", params["VirtualServerName"].(string)),
+					resource.TestCheckResourceAttr("data.vcd_lb_virtual_server.http", "ip_address", params["EdgeGatewayIp"].(string)),
+					resource.TestCheckResourceAttr("data.vcd_lb_virtual_server.http", "protocol", "http"),
+					resource.TestCheckResourceAttr("data.vcd_lb_virtual_server.http", "port", "8888"),
+					resource.TestMatchResourceAttr("data.vcd_lb_virtual_server.http", "app_profile_id", regexp.MustCompile(`^applicationProfile-\d*$`)),
+					resource.TestMatchResourceAttr("data.vcd_lb_virtual_server.http", "server_pool_id", regexp.MustCompile(`^pool-\d*$`)),
+					resource.TestCheckResourceAttr("data.vcd_lb_virtual_server.http", "app_rule_ids.#", "2"),
+					resource.TestMatchResourceAttr("data.vcd_lb_virtual_server.http", "app_rule_ids.0", regexp.MustCompile(`^applicationRule-\d*$`)),
+					resource.TestMatchResourceAttr("data.vcd_lb_virtual_server.http", "app_rule_ids.1", regexp.MustCompile(`^applicationRule-\d*$`)),
+				),
+			},
+
+			// Check that import works
+			resource.TestStep{
+				ResourceName:      "vcd_lb_virtual_server.virtual-server-import",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdByOrgVdcEdge(testConfig, params["VirtualServerName"].(string)),
+			},
+			// Simple check - without app profile, rule ids or server pool
```

Comment:
Seems that validates update? or not?

Reply:
Yes. Further steps are doing "updates" when resource names are left the same.

Reply:
Maybe we should state in comment

Reply:
Added comment for each step number